api/progress.go

```diff
@@ -0,0 +1,19 @@
+package api
+
+import "sync"
+
+// Progress is a generalized struct which can track the progress of a task
+type Progress struct {
+	sync.Mutex
```

Comment:
We do not need an explicit lock here.
Please consider using `[sync/atomic](https://pkg.go.dev/sync/atomic)` to atomically load the progress from the task scheduler and update the progress from the task executor.